### PR TITLE
KAA-780: [C SDK] Make the amount of parallel log upload batches configurable.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: client/client-multi/client-c/src/kaa/kaa_common_schema.h [hunk 1]
```diff
@@ -38,7 +38,7 @@ extern "C" {
 
 
 typedef void (*serialize_fn)(avro_writer_t writer, void *data);
-typedef void *(*deserialize_fn)(void);
+typedef void *(*deserialize_fn)(); // NOTE: Do not add void in the parameter list.
```
Comment (rasendubi) [inline client/client-multi/client-c/src/kaa/kaa_common_schema.h]: Shouldn't the function take `void *`?

File: client/client-multi/client-c/test/test_kaa_log.c [hunk 2]
```diff
@@ -69,6 +67,15 @@ static kaa_logger_t *logger = NULL;
 static kaa_status_t *status = NULL;
 static kaa_channel_manager_t *channel_manager = NULL;
 
+static kaa_service_t OPERATIONS_SERVICES[] = { KAA_SERVICE_PROFILE
```
Comment (rasendubi) [inline client/client-multi/client-c/test/test_kaa_log.c]: Shouldn't this be `const kaa_service_t`?

File: client/client-multi/client-c/src/kaa/kaa_common_schema.h (hunk 1 above)
Comment (ghost) [inline client/client-multi/client-c/src/kaa/kaa_common_schema.h]: deserialize_fn is a something like a parent class for deserialize_wo_ctx_fn and deserialize_w_ctx_fn. The signature of these typedefs of pointer to functions is different so we can't use void*

Maybe it isn't a good practice to use this kind of hack, but now to make it to more clear a lot of code need to be re-implemented in Avro Gen C.

File: client/client-multi/client-c/test/test_kaa_log.c (hunk 2 above)
Comment (ghost) [inline client/client-multi/client-c/test/test_kaa_log.c]: Sure, I will do.

File: client/client-multi/client-c/src/kaa/kaa_common_schema.h (hunk 1 above)
Comment (rasendubi) [inline client/client-multi/client-c/src/kaa/kaa_common_schema.h]: Could you put that in a comment or open a ticket, please?

Comment (ghost) [inline client/client-multi/client-c/src/kaa/kaa_common_schema.h]: Added a comment.
I suggest to discuss improvements of Avro C and we will open a story. Ok?

**Pull request merged**